From 5030f9916c2e71eaa690408b26a5f03fdc971913 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 01 Sep 2022 14:55:50 +0300
Subject: [PATCH] [SW-102283] gaudi2: use wd registers and gpio number from the specs

The watch dog gpio deassertion is done in firmware and also in
hl-fw-loader(non-secured update -r) hence the gpio and its registers
should be used from the compatibility layer i,e. from specs which
helps in ease of code maintenance.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I9498f865f56d91d24b78902200c3baa4e9fbdbae
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 8663138..c4430d5 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1015,7 +1015,6 @@
 	preboot_puts_raw("Outstanding transaction are zero, proceeding with reset");
 }
 
-#define WD_ASSERTION_GPIO	14
 #define FORCE_RECOVERY_MASK	BIT(2)
 void preboot_clear_wd_gpio(void)
 {
@@ -1032,11 +1031,11 @@
 	if (!val)
 		return;
 
-	mask = BIT(WD_ASSERTION_GPIO);
-	regmap_writel(mask, mmPSOC_GPIO3_OUTENSET);
-	val  = regmap_readl(mmPSOC_GPIO3_DATAOUT);
+	mask = BIT(GAUDI2_WD_GPIO);
+	regmap_writel(mask, mmWD_GPIO_OUTSET_REG);
+	val  = regmap_readl(mmWD_GPIO_DATAOUT_REG);
 	val |= mask;
-	regmap_writel(val, mmPSOC_GPIO3_DATAOUT);
+	regmap_writel(val, mmWD_GPIO_DATAOUT_REG);
 }
 
 /*
